### PR TITLE
Stm reader

### PR DESCRIPTION
**Pull request by RubelMozumder**

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -357,25 +361,23 @@ ptyprocess==0.7.0
...
+pycodestyle==2.10.0
     # via nexusutils (pyproject.toml)
 pycparser==2.21
     # via cffi
 pygments==2.13.0
     # via
     #   ipython
     #   nbconvert
-pylint==2.15.3
+pylint==2.17.2
```
Comment (sanbrock) [inline dev-requirements.txt:372]: why do we have to change all these dependencies?

File: nexusutils/dataconverter/readers/stm/SPM_data_folder/spm_test_output.test.nxs
Comment (sanbrock) [inline nexusutils/dataconverter/readers/stm/SPM_data_folder/spm_test_output.test.nxs:1]: why are these test files not under /test/...?

File: nexusutils/dataconverter/readers/stm/config_file.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+"/ENTRY[entry]/INSTRUMENT[instrument]/ELECTRONANALYSER[electronanalyser]/description": "@eln",
+"/ENTRY[entry]/INSTRUMENT[instrument]/ELECTRONANALYSER[electronanalyser]/name": "@xps_tocken:analyzer_info/name",
+"/ENTRY[entry]/INSTRUMENT[instrument]/ELECTRONANALYSER[electronanalyser]/ENERGYDISPERSION[energydispersion]/entrance_slit": "@xps_tocken:region/analyzer_slit",
```
Comment (sanbrock) [inline nexusutils/dataconverter/readers/stm/config_file.json:4]: how are these paths relate to STM or to pointing STM raw data to hdf5?

File: dev-requirements.txt (hunk 1 above)
Comment (RubelMozumder) [reply dev-requirements.txt:372]: while including 'nanonispy' and rerun pip-compile it has created the file itself satisfying  the dependencies condition.

**Comment by RubelMozumder** [comment]
I will create a new pull request later.

**Pull request closed without merging**